src/databases/postgresql.rs

```diff
@@ -181,6 +182,50 @@ pub fn remove_filtered_policy(
         })
 }
 
+pub(crate) fn save_policy (conn: Pool, rules: Vec<NewCasbinRule>) -> Result<()> {
+    use schema::casbin_rules::dsl::casbin_rules;
+
+    conn.transaction::<_, DieselError, _>(|| {
+        diesel::insert_into(casbin_rules)
+            .values(&rules)
+            .execute(&conn)
+            .map_err(|_| DieselError::RollbackTransaction)
+    })
+        .map(|_| ())
+        .map_err(|err| Box::new(Error::DieselError(err)) as Box<dyn StdError>)
+}
+
+pub(crate) fn load_policy(conn: Pool) -> Result<Vec<CasbinRule>> {
+    use schema::casbin_rules::dsl::casbin_rules;
+
+    casbin_rules
+        .load::<CasbinRule>(&conn)
+        .map_err(|err| Box::new(Error::DieselError(err)) as Box<dyn StdError>)
+}
+
+pub(crate) fn add_policy(conn: Pool, new_rule: NewCasbinRule) -> Result<bool> {
+    use schema::casbin_rules::dsl::casbin_rules;
+
+    diesel::insert_into(casbin_rules)
+        .values(&new_rule)
+        .execute(&conn)
```

Comment:
A small improvement which can be done by use is:

replace `execute` by `get_result`
so that we don't need to check the influenced rows. I should have done it but I didn't... could you do it?

Reply:
Yeah, I am working on it. Thx.